library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsMediaChunk.java

```diff
@@ -291,6 +299,27 @@ public boolean isLoadCompleted() {
     return loadCompleted;
   }
 
+  /**
+   * Return the indexes of the first samples from each sample queue for this media chunk
+   *
+   * @return sample index {@link SampleQueue#getWriteIndex()}
+   */
+  int [] getFirstSampleIndexes() {
+    return firstSampleIndexes;
+  }
+
+  /**
+   * Set the index of the first sample written to a sample queue from this media chunk,
+   * indexed by the caller's stream index for the sample queue
+   *
+   * @param streamIndex - caller's index for the {@link SampleQueue}
+   * @param firstSampleIndex - index value to store
+   */
+  void setFirstSampleIndex(int streamIndex, int firstSampleIndex) {
```

Comment:
We removed this method in the end because the values can generated locally by passing in the SampleQueues to the init() method. 

The case where a sample queue is created by this chunk can actually be removed entirely because if a sample queue doesn't exist yet, we know that it is created by this chunk (or even a later one), so the index to discard from is always 0.